sdk/servicebus/azure-servicebus/azure/servicebus/_common/message.py

```diff
@@ -809,7 +809,7 @@ def __init__(
     ) -> None:
         self._amqp_transport = kwargs.pop("amqp_transport", PyamqpTransport)
         super(ServiceBusReceivedMessage, self).__init__(None, message=message)  # type: ignore
-        if self._amqp_transport != PyamqpTransport:
+        if not issubclass(self._amqp_transport, PyamqpTransport):
```

Comment:
I thought we had a `type` attribute for doing these kinds of checks?

Reply:
oh yep we do! forgot 🤦